From 0ac7b45ec42ea83beea2eeb22d853de9c309e511 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jakab=20Krist=C3=B3f?= <jakab.kristof@balasys.hu>
Date: Fri, 29 Jul 2016 14:44:05 +0000
Subject: [PATCH 01/19] Add debug flavour

Differences between generic and debug:
* CONFIG_KASAN=y
* CONFIG_KASAN_INLINE=y
---
 debian.wily/config/amd64/config.flavour.debug      | 10 ++++++++++
 debian.wily/config/amd64/config.flavour.generic    |  1 +
 debian.wily/config/amd64/config.flavour.lowlatency |  1 +
 debian.wily/config/arm64/config.common.arm64       |  1 +
 debian.wily/config/armhf/config.common.armhf       |  1 +
 debian.wily/config/config.common.ubuntu            |  6 +++++-
 debian.wily/config/i386/config.common.i386         |  1 +
 debian.wily/config/powerpc/config.common.powerpc   |  1 +
 debian.wily/config/ppc64el/config.common.ppc64el   |  1 +
 debian.wily/control.d/vars.debug                   |  6 ++++++
 10 files changed, 28 insertions(+), 1 deletion(-)
 create mode 100644 debian.wily/config/amd64/config.flavour.debug
 create mode 100644 debian.wily/control.d/vars.debug

diff --git a/debian.wily/config/amd64/config.flavour.debug b/debian.wily/config/amd64/config.flavour.debug
new file mode 100644
index 0000000..06a660d
--- /dev/null
+++ b/debian.wily/config/amd64/config.flavour.debug
@@ -0,0 +1,10 @@
+#
+# Config options for config.flavour.debug automatically generated by splitconfig.pl
+#
+CONFIG_HZ=250
+# CONFIG_HZ_1000 is not set
+CONFIG_HZ_250=y
+# CONFIG_IRQ_FORCED_THREADING_DEFAULT is not set
+CONFIG_KASAN=y
+# CONFIG_PREEMPT is not set
+CONFIG_PREEMPT_VOLUNTARY=y
diff --git a/debian.wily/config/amd64/config.flavour.generic b/debian.wily/config/amd64/config.flavour.generic
index 5c2e4e1..aa840d5 100644
--- a/debian.wily/config/amd64/config.flavour.generic
+++ b/debian.wily/config/amd64/config.flavour.generic
@@ -5,5 +5,6 @@ CONFIG_HZ=250
 # CONFIG_HZ_1000 is not set
 CONFIG_HZ_250=y
 # CONFIG_IRQ_FORCED_THREADING_DEFAULT is not set
+# CONFIG_KASAN is not set
 # CONFIG_PREEMPT is not set
 CONFIG_PREEMPT_VOLUNTARY=y
diff --git a/debian.wily/config/amd64/config.flavour.lowlatency b/debian.wily/config/amd64/config.flavour.lowlatency
index 6b121e3..ab3f1a8 100644
--- a/debian.wily/config/amd64/config.flavour.lowlatency
+++ b/debian.wily/config/amd64/config.flavour.lowlatency
@@ -5,5 +5,6 @@ CONFIG_HZ=1000
 CONFIG_HZ_1000=y
 # CONFIG_HZ_250 is not set
 CONFIG_IRQ_FORCED_THREADING_DEFAULT=y
+# CONFIG_KASAN is not set
 CONFIG_PREEMPT=y
 # CONFIG_PREEMPT_VOLUNTARY is not set
diff --git a/debian.wily/config/arm64/config.common.arm64 b/debian.wily/config/arm64/config.common.arm64
index f6c3a38..428e746 100644
--- a/debian.wily/config/arm64/config.common.arm64
+++ b/debian.wily/config/arm64/config.common.arm64
@@ -59,6 +59,7 @@ CONFIG_IOMMU_IO_PGTABLE_LPAE=y
 # CONFIG_IPMI_SI_PROBE_DEFAULTS is not set
 # CONFIG_IRQ_FORCED_THREADING_DEFAULT is not set
 CONFIG_JUMP_LABEL=y
+# CONFIG_KASAN is not set
 CONFIG_KVM=y
 CONFIG_LEDS_TRIGGER_HEARTBEAT=m
 CONFIG_LOG_BUF_SHIFT=14
diff --git a/debian.wily/config/armhf/config.common.armhf b/debian.wily/config/armhf/config.common.armhf
index 168a328e..26b5c11 100644
--- a/debian.wily/config/armhf/config.common.armhf
+++ b/debian.wily/config/armhf/config.common.armhf
@@ -51,6 +51,7 @@ CONFIG_IMA=y
 # CONFIG_IPMI_SI_PROBE_DEFAULTS is not set
 # CONFIG_IRQ_FORCED_THREADING_DEFAULT is not set
 # CONFIG_JUMP_LABEL is not set
+# CONFIG_KASAN is not set
 CONFIG_KVM=y
 CONFIG_LEDS_TRIGGER_HEARTBEAT=y
 CONFIG_LOG_BUF_SHIFT=17
diff --git a/debian.wily/config/config.common.ubuntu b/debian.wily/config/config.common.ubuntu
index 6810269..f01a3a8 100644
--- a/debian.wily/config/config.common.ubuntu
+++ b/debian.wily/config/config.common.ubuntu
@@ -1374,6 +1374,7 @@ CONFIG_CONFIGFS_FS=m
 CONFIG_CONNECTOR=y
 CONFIG_CONSOLE_POLL=y
 CONFIG_CONSOLE_TRANSLATIONS=y
+CONFIG_CONSTRUCTORS=y
 CONFIG_CONTEXT_SWITCH_TRACER=y
 CONFIG_COPS=m
 CONFIG_COPS_DAYNA=y
@@ -3836,7 +3837,9 @@ CONFIG_K3_DMA=m
 CONFIG_KALLSYMS=y
 CONFIG_KALLSYMS_ALL=y
 CONFIG_KARMA_PARTITION=y
-# CONFIG_KASAN is not set
+CONFIG_KASAN_INLINE=y
+# CONFIG_KASAN_OUTLINE is not set
+CONFIG_KASAN_SHADOW_OFFSET=0xdffffc0000000000
 CONFIG_KDB_CONTINUE_CATASTROPHIC=0
 CONFIG_KDB_DEFAULT_ENABLE=0x1
 CONFIG_KDB_KEYBOARD=y
@@ -7664,6 +7667,7 @@ CONFIG_TELCLOCK=m
 CONFIG_TEST_BPF=m
 CONFIG_TEST_FIRMWARE=m
 CONFIG_TEST_HEXDUMP=m
+# CONFIG_TEST_KASAN is not set
 CONFIG_TEST_KSTRTOX=m
 # CONFIG_TEST_LIST_SORT is not set
 CONFIG_TEST_LKM=m
diff --git a/debian.wily/config/i386/config.common.i386 b/debian.wily/config/i386/config.common.i386
index 9c63051..c0d6f24 100644
--- a/debian.wily/config/i386/config.common.i386
+++ b/debian.wily/config/i386/config.common.i386
@@ -57,6 +57,7 @@ CONFIG_IMA=y
 CONFIG_IPMI_SI_PROBE_DEFAULTS=y
 CONFIG_ISA=y
 CONFIG_JUMP_LABEL=y
+# CONFIG_KASAN is not set
 CONFIG_KVM=m
 CONFIG_LEDS_TRIGGER_HEARTBEAT=m
 CONFIG_LOG_BUF_SHIFT=17
diff --git a/debian.wily/config/powerpc/config.common.powerpc b/debian.wily/config/powerpc/config.common.powerpc
index 6427411..559cb66d 100644
--- a/debian.wily/config/powerpc/config.common.powerpc
+++ b/debian.wily/config/powerpc/config.common.powerpc
@@ -47,6 +47,7 @@ CONFIG_IPMI_SI_PROBE_DEFAULTS=y
 # CONFIG_IRQ_FORCED_THREADING_DEFAULT is not set
 # CONFIG_ISA is not set
 CONFIG_JUMP_LABEL=y
+# CONFIG_KASAN is not set
 CONFIG_KVM=y
 CONFIG_LEDS_TRIGGER_HEARTBEAT=m
 CONFIG_LOG_BUF_SHIFT=17
diff --git a/debian.wily/config/ppc64el/config.common.ppc64el b/debian.wily/config/ppc64el/config.common.ppc64el
index 5168d31..2f0a2db 100644
--- a/debian.wily/config/ppc64el/config.common.ppc64el
+++ b/debian.wily/config/ppc64el/config.common.ppc64el
@@ -68,6 +68,7 @@ CONFIG_IPMI_SI_PROBE_DEFAULTS=y
 CONFIG_IRQ_ALL_CPUS=y
 # CONFIG_IRQ_FORCED_THREADING_DEFAULT is not set
 CONFIG_JUMP_LABEL=y
+# CONFIG_KASAN is not set
 CONFIG_KERNEL_START=0xc000000000000000
 CONFIG_KVM=y
 CONFIG_LEDS_TRIGGER_HEARTBEAT=m
diff --git a/debian.wily/control.d/vars.debug b/debian.wily/control.d/vars.debug
new file mode 100644
index 0000000..2b264d4
--- /dev/null
+++ b/debian.wily/control.d/vars.debug
@@ -0,0 +1,6 @@
+arch="amd64"
+supported="Debug"
+target="Geared toward desktop and server systems."
+desc="=HUMAN= SMP"
+bootloader="grub-pc [i386 amd64 x32] | grub-efi-amd64 [amd64 x32] | grub-efi-ia32 [i386 amd64 x32] | grub [i386 amd64 x32] | lilo (>= 19.1) [i386 amd64 x32] | flash-kernel [armhf arm64] | grub-ieee1275 [ppc64el]"
+provides="kvm-api-4, redhat-cluster-modules, ivtv-modules, virtualbox-guest-modules [i386 amd64 x32]"
-- 
2.9.2

